test/e2e/config/ironic.yaml

```diff
@@ -14,11 +16,23 @@ images:
 
 # These variables can be overridden with environment variables.
 variables:
+  # These settings are for the "default" cluster, which is used for all tests except upgrade ones
   DEPLOY_IRONIC: "true"
   DEPLOY_BMO: "true"
   DEPLOY_CERT_MANAGER: "true"
   BMO_KUSTOMIZATION: "../../config/overlays/e2e"
   IRONIC_KUSTOMIZATION: "../../ironic-deployment/overlays/e2e"
+  KUBECONFIG_DEFAULT_CONTEXT_NAME: "bmo-e2e"
+
+  # These settings are for the separate cluster that are used for upgrade tests
+  UPGRADE_DEPLOY_IRONIC: "true"
+  UPGRADE_DEPLOY_BMO: "true"
+  UPGRADE_DEPLOY_CERT_MANAGER: "true"
+  BMO_UPGRADE_FROM_KUSTOMIZATION: "../../config/overlays/e2e-release-0.4"
+  BMO_RELEASE_BRANCH: "release-0.4"
+  UPGRADE_IRONIC_KUSTOMIZATION: "../../ironic-deployment/overlays/e2e-upgrade"
```

Comment:
Hmm. I didn't want to use `IRONIC_UPGRADE_FROM_KUSTOMIZATION`, because we (for now) are not upgrading ironic in this test. And using the same one that we use for the main cluster doesn't work either, since we use different ip for the baremetal network (which changes the ip used by ironic, defined in `ironic.env`).

The name `UPGRADE_IRONIC_KUSTOMIZATION` means that it's the same as `IRONIC_KUSTOMIZATION`, but for the upgrade cluster. Do you think `IRONIC_UPGRADE_FROM_KUSTOMIZATION` is acceptable and better here?

Reply:
> And using the same one that we use for the main cluster doesn't work either, since we use different ip for the baremetal network (which changes the ip used by ironic, defined in ironic.env).

Good point, I did not think about this. But how does it work then when we upgrade BMO? Don't we switch to the other Ironic then?

Reply:
> Good point, I did not think about this. But how does it work then when we upgrade BMO? Don't we switch to the other Ironic then?

Ah yes. I didn't think about that either. It does look like it switches to the other ironic :laughing: Let's see if I could find a way around this env file.